### PR TITLE
Feat/activity v2

### PR DESCRIPTION
**Pull request by dlipicar**
Part of https://github.com/status-im/status-desktop/issues/16275
status-go part: https://github.com/status-im/status-go/pull/6102

### What does the PR do

- Adapt to new ActivityV2 API
- Reworked activity context menu
- Hide Activity filter
- Hide Collectibles details activity tab

Some work was left out of the initial iteration: 
- Support for some activity types (Community tokens/collectibles txs, DApps)
- Repeat transaction option in context menu

### Affected areas

Activity, Collectibles Details

### Architecture compliance

- [ ] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

https://github.com/user-attachments/assets/032c1ad4-dbbc-4995-9454-65209cca3332

- [ ] I've checked the design and this PR matches it

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/views/RightTabView.qml [hunk 1]
```diff
@@ -149,6 +149,7 @@ RightTabBaseView {
                     icon.color: checked ? Theme.palette.primaryColor1 : Theme.palette.baseColor1
                     Behavior on icon.color { ColorAnimation { duration: 200; easing.type: Easing.InOutQuad } }
                     highlighted: checked
+                    visible: walletTabBar.currentIndex !== 2 // TODO #16761: Re-enable filter for activity when implemented
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/views/RightTabView.qml:152]: Magic numbers?

File: ui/imports/shared/views/TransactionContextMenu.qml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    id: root
+
+    required property string networkExplorerName
+
+    signal viewTxOnExplorerRequested
+    signal copyTxHashRequested
+
+    StatusAction {
+        objectName: "viewTxOnExplorerItem"
+        text: qsTr("View on ") + networkExplorerName
```
Comment (caybro) [inline ui/imports/shared/views/TransactionContextMenu.qml:15]:
```suggestion
        text: qsTr("View on %1").arg(networkExplorerName)
```

Comment (caybro) [reply ui/imports/shared/views/TransactionContextMenu.qml:15]: Never cut sentences like this pls

**Review by caybro — COMMENTED** [review]
Minor stuff, otherwise LGTM

File: ui/imports/shared/views/TransactionContextMenu.qml (hunk 2 above)
Comment (dlipicar) [reply ui/imports/shared/views/TransactionContextMenu.qml]: ah crap, working for too long in status-go does this to a man
fixed here https://github.com/status-im/status-desktop/pull/16762/commits/a0e683ea98bcc94d0aad56ca283172a2069fdf3e

**Review by caybro — APPROVED** [review]
LGTM, nice!

File: ui/imports/shared/controls/TransactionDelegate.qml [hunk 3]
```diff
@@ -716,7 +396,7 @@ StatusListItem {
         case Constants.TransactionType.Approve:
             return failed ? qsTr("Failed to set spending cap") : (isPending ? qsTr("Setting spending cap") : qsTr("Spending cap set"))
         default:
-            return ""
+            return "Interaction"
```
Comment (caybro) [inline ui/imports/shared/controls/TransactionDelegate.qml:399]:
```suggestion
            return qsTr("Interaction")
```

File: storybook/stubs/AppLayouts/Wallet/stores/RootStore.qml
Comment (caybro) [inline storybook/stubs/AppLayouts/Wallet/stores/RootStore.qml:1]: The stub should ideally be empty, and the impl moved to the respective SB page. Although I agree that if you need the same mock functionality in multiple places (like SB and a QML test), it's better than duplicating it

File: ui/imports/utils/Utils.qml [hunk 4]
```diff
@@ -644,6 +644,44 @@ QtObject {
         return getEtherscanUrl(networkShortName, testnetMode, tx, false /* is TX */)
     }
 
+    // Get Chain Explorer name
+    function getChainExplorerName(networkShortName) {
+        if (networkShortName === Constants.networkShortChainNames.arbitrum) {
+            return qsTr("Arbiscan")
+        }
+        if (networkShortName === Constants.networkShortChainNames.optimism) {
+            return qsTr("Optimistic")
```
Comment (micieslak) [inline ui/imports/utils/Utils.qml:653]:
```suggestion
            return qsTr("Optimism")
```

File: storybook/stubs/AppLayouts/Wallet/stores/RootStore.qml [hunk 5]
```diff
@@ -35,4 +38,18 @@ QtObject {
     function getOpenSeaCollectibleUrl(networkShortName, contractAddress, tokenId) {
         return "https://somedummyurl.com"
     }
+
+    function getDappDetails(chainId, contractAddress) {
```
Comment (micieslak) [inline storybook/stubs/AppLayouts/Wallet/stores/RootStore.qml:42]: As @caybro mentioned stub should be empty file. But here the problem is with singleton store. It should be fixed but ofc not as a part of this pr. We can live with this for now.

File: storybook/stubs/AppLayouts/Wallet/stores/RootStore.qml [hunk 6]
```diff
@@ -3,6 +3,9 @@ pragma Singleton
 
 import QtQml 2.15
 
+import utils 1.0
+import Models 1.0
```
Comment (micieslak) [inline storybook/stubs/AppLayouts/Wallet/stores/RootStore.qml:7]: is it necessary?

File: ui/imports/shared/controls/TransactionDelegate.qml
Comment (micieslak) [inline ui/imports/shared/controls/TransactionDelegate.qml:1]: In the future we should aim for making this delegate simple-ui only component, with no dependency on stores. All what is needed for the UI should be prepared earlier.

**Review by micieslak — COMMENTED** [review]
LGTM in general, minor comments only.

File: ui/imports/shared/controls/TransactionDelegate.qml
Comment (alexjba) [inline ui/imports/shared/controls/TransactionDelegate.qml:1]: This is quite a large delegate. Some qml tests would be nice here.

**Review by alexjba — COMMENTED** [review]
LGTM!

One question below.

File: ui/imports/shared/controls/TransactionDelegate.qml
Comment (dlipicar) [reply ui/imports/shared/controls/TransactionDelegate.qml:1]: I promise there will be a rework for this component, just not for 2.32!

File: storybook/stubs/AppLayouts/Wallet/stores/RootStore.qml (hunk 6 above)
Comment (dlipicar) [reply storybook/stubs/AppLayouts/Wallet/stores/RootStore.qml:7]: oh I was taking `ModelsData.icons.socks` from there

File: ui/imports/utils/Utils.qml (hunk 4 above)
Comment (dlipicar) [reply ui/imports/utils/Utils.qml:653]: image
Figma says this is called Optimistic :D

**Review by alexjba — APPROVED** [review]

File: ui/imports/utils/Utils.qml (hunk 4 above)
Comment (caybro) [reply ui/imports/utils/Utils.qml:653]: Yeah it is correct

**Comment by virginiabalducci** [comment]
Activity has been tested for the following (except Failed statuses)

SD-315 - Send Asset (ETH / ERC20)
SD-316 - Sent collectible (ERC721/ERC1155)
SD-317 - Bridge Asset
SD-318 - Swap Asset
SD-319 - Approval for Asset
All Accounts vs Specific Account Activity List Items

1 issue found so far:  https://github.com/status-im/status-desktop/issues/16788

Need instructions on how to test 
SD-342 - Unidentified transaction

**Comment by dlipicar** [comment]
@virginiabalducci bug fixed in latest commit

image

**Review by virginiabalducci — APPROVED** [review]

**Pull request merged**